### PR TITLE
Add convenience method for transforming to void

### PR DESCRIPTION
**Pull request by 0xTim**
Allows you to do

```swift
myFuture.transform(to: Future.void)
```

Moved from vapor-community/async#55

### CODE REVIEW

File: Sources/Async/Future+Void.swift [hunk 1]
```diff
@@ -13,3 +13,12 @@ extension Future where T == Void {
...
 }
+
+extension Future where T == Void {
+    /// Convenience function for returning void inside a future
+    public static var void: Void {
+        return _void
+    }
+}
+
+private let _void: Void = {}()
```
Comment (MrMage) [inline Sources/Async/Future+Void.swift]: How about just `private let _void: Void = {}()` instead, or simply `return ()` above? In this case, I don't think we are gaining anything from moving the value to a shared constant.

Comment (0xTim) [reply Sources/Async/Future+Void.swift]: Quite possibly not!

**Review by tanner0101 — APPROVED** [review]
This LGTM. I'm interested to know, does this support leading-dot syntax?

**Comment by 0xTim** [comment]
As in `.transform(to: .void)`?

**Comment by tanner0101** [comment]
Yeah, does that work?

**Comment by 0xTim** [comment]
No compiler doesn't appear to be able to infer it unfortunately 😞

**Comment by tanner0101** [comment]
@0xTim I wonder if we can get it to work by adding an overload like:
```swift
extension Future {
    public func transform(to void: Void) -> Future<Void> {
        return transform(to: void)
    }
}
extension Void {
    public static var void: Void { return () } 
}
```

**Comment by tanner0101** [comment]
Wait, actually now that I look at this, won't this cause nested `Future`s? 🤔 

`transform(to: ...)` should usually accept a non-Future value.

**Comment by tanner0101** [comment]
Hmm, can't extend Void. Something like this would work though:

```swift
extension Future {
    public func transform(to void: VoidType) -> Future<Void> {
        return transform(to: ())
    }
}

public enum VoidType {
    case void
}
```

```swift
let future: Future<String>
let void: Future<Void> = future.transform(to: .void)
```

The extension on `Future` to add a static `Void` seems kind of confusing the more I look at it.

**Comment by tanner0101** [comment]
Or also:

```swift
public let void: Void = ()
```

```swift
let future: Future<String>
let void: Future<Void> = future.transform(to: void)
```

**Comment by MrMage** [comment]
At that point, why not just `.transform(to: ())`? I think it's not less clear than the other options.

**Comment by 0xTim** [comment]
I prefer `.transform(to: .void)`, I think it's more readable to `()` but I'm happy to do it manually/just get used to it if it gets too confusing!

**Comment by MrMage** [comment]
Agreed, () looks a bit weird.

It seems like we have to jump through surprisingly many hoops to accomplish this — what would you think of a dedicated `toVoid()` or `transformToVoid()` method instead? In fact, that might even open up extra avenues for optimization (might be a bit far-fetched, though).

**Comment by tanner0101** [comment]
Closing this due to inactivity / lack of consensus. Feel free to re-open if you want to continue to discuss.

**Pull request closed without merging**